onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -131,17 +131,24 @@ static Status GetExternalDataInfo(const ONNX_NAMESPACE::TensorProto& tensor_prot
   std::unique_ptr<onnxruntime::ExternalDataInfo> external_data_info;
   ORT_RETURN_IF_ERROR(onnxruntime::ExternalDataInfo::Create(tensor_proto.external_data(), external_data_info));
 
-  if (tensor_proto_dir != nullptr) {
-    external_file_path = onnxruntime::ConcatPathComponent<ORTCHAR_T>(tensor_proto_dir, external_data_info->GetRelPath());
+  const auto& location = external_data_info->GetRelPath();
+
+  if (location == onnxruntime::utils::kMemoryAddressTag) {
+    external_file_path = location;
+    tensor_byte_size = external_data_info->GetLength();
   } else {
-    external_file_path = external_data_info->GetRelPath();
+    if (tensor_proto_dir != nullptr) {
+      external_file_path = onnxruntime::ConcatPathComponent<ORTCHAR_T>(tensor_proto_dir, external_data_info->GetRelPath());
+    } else {
+      external_file_path = external_data_info->GetRelPath();
+    }
+
+    ORT_RETURN_IF_ERROR(onnxruntime::utils::GetSizeInBytesFromTensorProto<0>(tensor_proto, &tensor_byte_size));
```

Comment:
why do we not need to check the external data info's size with the computed size if using kMemoryAddressTag?

Reply:
No reason. Dims should be valid in either case and this should be done in both cases.